python/server/qgsserverrequest.sip

```diff
@@ -0,0 +1,85 @@
+/***************************************************************************
+                          qgsserverrequest.h
+
+  Define request class for getting request contents
+  -------------------
+  begin                : 2016-12-05
+  copyright            : (C) 2016 by David Marteau
+  email                : david dot marteau at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+/**
+ * \ingroup server
+ * QgsServerRequest
+ * Class defining request interface passed to services QgsService::executeRequest() method
+ */
+
+class QgsServerRequest
+{
+%TypeHeaderCode
+#include "qgsserverrequest.h"
+%End
+  public:
+
+    enum Method
+    {
+      HeadMethod, PutMethod, GetMethod, PostMethod, DeleteMethod
+    };
+
+    /**
+     * Constructor
+     *
+     * @param url the url string
+     * @param method the request method
+     */
+    QgsServerRequest( const QString& url, Method method );
+
+    /**
+     * Constructor
+     *
+     * @param url QUrl
+     * @param method the request method (default to GetMethod)
+     */
+    QgsServerRequest( const QUrl& url, Method method = GetMethod );
+
+    //! destructor
+    virtual ~QgsServerRequest();
+
+    /**
+     * @return the value of the header field for that request
+     */
+    virtual QString getHeader( const QString& name ) const;
+
+    /**
+     * @return the request url
+     */
+    QUrl url() const;
+
+    /**
+     * @return the request method
+     */
+    Method method() const;
+
+    /**
+     *      * @return query params
+     *           */
+    QMap<QString, QString> parameters() const;
```

Comment:
> you always have to convert your parameter keys and values

I don't agree with this sentence, we usually do that in the OGC service because not every body respect exactly the spec, than we do that to increase the compatibility. Than I think that we should be able to write a service that don't do that.

For the initial question I understand your position ant it's true that's will be easier :-)